nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-services/src/main/java/org/apache/nifi/mongodb/MongoDBControllerService.java

```diff
@@ -105,7 +110,18 @@ public final void closeClient() {
     }
 
     protected String getURI(final ConfigurationContext context) {
-        return context.getProperty(URI).evaluateAttributeExpressions().getValue();
+        final String _uri = context.getProperty(URI).evaluateAttributeExpressions().getValue();
+        final String user = context.getProperty(DB_USER).evaluateAttributeExpressions().getValue();
+        final String passw = context.getProperty(DB_PASSWORD).evaluateAttributeExpressions().getValue();
+        if (!_uri.contains("@") && user != null && passw != null) {
+            try {
+                return _uri.replaceFirst("://", "://" + URLEncoder.encode(user, StandardCharsets.UTF_8.toString()) + ":" + URLEncoder.encode(passw, StandardCharsets.UTF_8.toString()) + "@");
+            } catch (final UnsupportedEncodingException e) {
```

Comment:
Probably worth logging the exception here, no? At least at debug level, but maybe warning level makes sense.

Reply:
I agree with a WARM, but I think it will never be triggered.